lib/events/emitter.go

```diff
@@ -168,6 +170,37 @@ func (*DiscardEmitter) ResumeAuditStream(ctx context.Context, sid session.ID, up
 	return &DiscardStream{}, nil
 }
 
+// NewWriterEmitter returns a new instance of emitter writing to writer
+func NewWriterEmitter(w io.WriteCloser) *WriterEmitter {
+	return &WriterEmitter{
+		w:         w,
+		WriterLog: NewWriterLog(w),
+	}
+}
+
+// WriterEmitter is an emitter that emits all events
+// to the external writer
+type WriterEmitter struct {
+	w io.WriteCloser
+	*WriterLog
```

Comment:
what is the `WriterLog` here for?
`WriterEmitter.EmitAuditEvent` doesn't seem to call it.

Related - what is the difference between `WriterLog` and `WriterEmitter`?

Reply:
It's to preserve backwards compatibility for now and avoid huge refactor in 4.4. I will try to make a better PR in 5.0 without those things, but it's purely to comply to interfaces